src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -21,40 +21,42 @@ internal static partial class Sys
         /// Returns the number of bytes placed into the buffer on success; bufferSize if the buffer is too small; and -1 on error.
         /// </returns>
         [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_ReadLink", SetLastError = true)]
-        private static partial int ReadLink(ref byte path, byte[] buffer, int bufferSize);
+        private static unsafe partial int ReadLink(ref byte path, byte* buffer, int bufferSize);
 
         /// <summary>
         /// Takes a path to a symbolic link and returns the link target path.
         /// </summary>
         /// <param name="path">The path to the symlink.</param>
         /// <returns>Returns the link to the target path on success; and null otherwise.</returns>
-        internal static string? ReadLink(ReadOnlySpan<char> path)
+        internal static unsafe string? ReadLink(ReadOnlySpan<char> path)
         {
             int outputBufferSize = 1024;
 
             // Use an initial buffer size that prevents disposing and renting
             // a second time when calling ConvertAndTerminateString.
             using var converter = new ValueUtf8Converter(stackalloc byte[1024]);
+            byte* stackBuffer = stackalloc byte[outputBufferSize];
```

Comment:
```suggestion
            Span<byte> stackBuffer = stackalloc byte[outputBufferSize];
```

Update the `ReadLink` function would make changing this natural.